cli/src/main.rs

```diff
@@ -131,38 +141,86 @@ fn run() -> Result<(), CliError> {
                 (@arg key: -k --key +takes_value "Signing key name")
             )
         )
-    ).get_matches();
-
-    let (batch_link, mut wait) = if let Some(upload_matches) = matches.subcommand_matches("upload")
-    {
-        upload(upload_matches)?
-    } else if let Some(exec_matches) = matches.subcommand_matches("exec") {
-        execute(exec_matches)?
-    } else if let Some(ns_matches) = matches.subcommand_matches("ns") {
-        namespace_registry(ns_matches)?
-    } else if let Some(perm_matches) = matches.subcommand_matches("perm") {
-        namespace_permission(perm_matches)?
-    } else if let Some(cr_matches) = matches.subcommand_matches("cr") {
-        contract_registry(cr_matches)?
-    } else if let Some(sp_matches) = matches.subcommand_matches("sp") {
-        smart_permission(sp_matches)?
+    );
+
+    let app = app.subcommand(
+        SubCommand::with_name("contract")
+            .about("List or show a Sabre smart contract")
+            .setting(AppSettings::SubcommandRequiredElseHelp)
+            .subcommand(
+                SubCommand::with_name("list")
+                    .about("List all registered Sabre smart contracts")
+                    .args(&[
+                        Arg::with_name("url")
+                            .help("URL to the Sawtooth REST API")
+                            .short("U")
+                            .long("url")
+                            .takes_value(true),
+                        Arg::with_name("format")
+                            .help("Format to display list of smart contracts in")
+                            .short("f")
+                            .long("format")
+                            .takes_value(true)
+                            .possible_values(&["human", "csv"])
+                            .default_value("human"),
+                    ]),
+            )
+            .subcommand(
+                SubCommand::with_name("show")
+                    .about("Show details about a registered Sabre smart contract")
+                    .args(&[
+                        Arg::with_name("url")
+                            .help("URL to the scabbard REST API")
```

Comment:
Should this say 'Sawtooth' instead of scabbard?

Reply:
yes, updated